### ISSUE TITLE
xUnit1009 should take [Optional] parameters into account

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#1638]
The xUnit1009 rule ("InlineData values must match the number of method parameters") should take `[Optional]` parameters into account, since they are supported. Note that this usage is slightly different than having parameters with declared optional values.

So this works (as expected):

```csharp
[Theory]
[InlineData]
public void TwoOptionalAttributes_NonePassed(object x = null, int y = 0)
{
    Assert.Null(x);
    Assert.Equal(0, y);
}
```

But this does not (and it should):

```csharp
[Theory]
[InlineData]
public void TwoOptionalAttributes_NonePassed([Optional]object x, [Optional]int y)
{
    Assert.Null(x);
    Assert.Equal(0, y);
}
```

**Comment by wachulski** [comment]
Just to be sure, this is invalid: `([Optional]object x, int y)` and all the following are valid:

`[InlineData]`
1. `object x, [Optional] int y = 0`
2. `object x = null, [Optional] int y = 0`
3. `[Optional]object x = null, [Optional] int y = 0`
4. `[Optional] object x = null, int y = 0`

**Comment by bradwilson** [comment]
Actually, none of those are valid.

You can't use `[Optional]` and also provide an optional value. The compiler refuses to let you do so.

**Comment by bradwilson** [comment]
```
error CS1745: Cannot specify default parameter value in conjunction with DefaultParameterAttribute or OptionalAttribute
```

**Comment by wachulski** [comment]
I thought `[Optional]` came from xunit. Anyway, thx for teaching me the basics.

**Comment by wachulski** [comment]
Ok, I'll take it 🙂

**Comment by wachulski** [comment]
Does xunit support it from its early days or since a specific version (like 2.2 for default parameter values in `Theory`)?

**Comment by bradwilson** [comment]
It was first available in 2.2. https://github.com/xunit/xunit/commit/d9e3e4c10011cd76574a006214bab51093c4c1e4

**Comment by wachulski** [comment]
All right then, I need to account for that.
